### PR TITLE
speed up several Piranha series functions

### PR DESCRIPTION
**Pull request by rwst**
The sincos benchmark shows an 8x speedup at N=1000 with this.

### CODE REVIEW

**Comment by certik** [comment]
+1

File: symengine/series.h [hunk 1]
```diff
@@ -222,8 +222,10 @@ class SeriesBase : public SeriesCoeffInterface {
         if (s == var) {
             //! fast atan(x)
             short sign = 1;
+            Poly monom(var), vsquare(var*var);
             for (unsigned int i = 1; i < prec; i += 2, sign *= -1) {
-                res_p += Series::pow(var, i, prec) * (Coeff(sign) / Coeff(i));
+                res_p += monom * (Coeff(sign) / Coeff(i));
+                monom *= vsquare;
```
Comment (isuruf) [inline symengine/series.h]: Shouldn't this be done in a truncated manner?

Comment (rwst) [inline symengine/series.h]: The degree of `monom` and `res_p` is always `< prec`.

**Comment by isuruf** [comment]
+1

**Pull request merged**